### DISCUSSION TITLE
Why code doesn't work not on a local blockchain

### DISCUSSION THREAD

**Discussion by blnkoff** [Elnaril/uniswap-universal-router-decoder#80] [Q&A]
Hello. I tried to use your code for buying and selling tokens. If I use it on local blockchain, it only works for buying tokens (not selling). If I use this code not on local blockchain, nothing works.I try to use it on BOB chain. All transactions are sent, but with status 'execution reverted'. It's example of transaction

https://explorer.gobob.xyz/tx/0x2c34d8251393089be2ec0513818be9ec304dfed95d8751204e6b78a93ed0f755

**Comment by Elnaril** [comment]
Hi again! :)

Like before, your question is way too broad. I cannot help if you don't investigate it deeper first and come with a very **specific** issue that concerns **directly** this library.

Having said that, please note:
- This lib has been developed for Ethereum. It works also for compatible blockchains where the UR and Permit2 contracts have been deployed with **the exact same code**.  And I'm not aware that Uniswap has ever officially deployed their contracts on the blockchain you mentioned ...
- The Permit2 usual address is not in the UR constructor parameters: have you used the right one when building your transaction ?
